query/stdlib/experimental/to.go

```diff
@@ -157,3 +166,309 @@ func newToProcedure(qs flux.OperationSpec, a plan.Administration) (plan.Procedur
+	var err error
+
+	var orgID platform.ID
+	var org string
+	// Get organization name and ID
+	if spec.Spec.Org != "" {
+		oID, ok := deps.OrganizationLookup.Lookup(ctx, spec.Spec.Org)
+		if !ok {
+			return nil, fmt.Errorf("failed to look up organization %q", spec.Spec.Org)
+		}
+		orgID = oID
+		org = spec.Spec.Org
+	} else if spec.Spec.OrgID != "" {
+		if oid, err := platform.IDFromString(spec.Spec.OrgID); err != nil {
+			return nil, err
+		} else {
+			orgID = *oid
+		}
+	} else {
+		// No org or orgID provided as an arg, use the orgID from the context
+		req := query.RequestFromContext(ctx)
+		if req == nil {
+			return nil, errors.New("missing request on context")
+		}
+		orgID = req.OrganizationID
+	}
+	if org == "" {
+		org = deps.OrganizationLookup.LookupName(ctx, orgID)
+		if org == "" {
+			return nil, fmt.Errorf("failed to look up organization name for ID %q", orgID.String())
+		}
+	}
+
+	var bucketID *platform.ID
+	var bucket string
+	// Get bucket name and ID
+	// User will have specified exactly one in the ToOpSpec.
+	if spec.Spec.Bucket != "" {
+		bID, ok := deps.BucketLookup.Lookup(ctx, orgID, spec.Spec.Bucket)
+		if !ok {
+			return nil, fmt.Errorf("failed to look up bucket %q in org %q", spec.Spec.Bucket, spec.Spec.Org)
+		}
+		bucketID = &bID
+		bucket = spec.Spec.Bucket
+	} else {
+		if bucketID, err = platform.IDFromString(spec.Spec.BucketID); err != nil {
+			return nil, err
+		}
+		bucket = deps.BucketLookup.LookupName(ctx, orgID, *bucketID)
+		if bucket == "" {
+			return nil, fmt.Errorf("failed to look up bucket with ID %q in org %q", bucketID, org)
+		}
+	}
+	return &ToTransformation{
+		ctx:      ctx,
+		bucket:   bucket,
+		bucketID: *bucketID,
+		org:      org,
+		orgID:    orgID,
+		d:        d,
+		cache:    cache,
+		spec:     spec.Spec,
+		deps:     deps,
+		buf:      storage.NewBufferedPointsWriter(influxdb.DefaultBufferSize, deps.PointsWriter),
+	}, nil
+}
+
+// Process does the actual work for the ToTransformation.
+func (t *ToTransformation) Process(id execute.DatasetID, tbl flux.Table) error {
+	return t.writeTable(t.ctx, tbl)
+}
+
+// UpdateWatermark updates the watermark for the transformation for the `to` flux function.
+func (t *ToTransformation) UpdateWatermark(id execute.DatasetID, pt execute.Time) error {
+	return t.d.UpdateWatermark(pt)
+}
+
+// UpdateProcessingTime updates the processing time for the transformation for the `to` flux function.
+func (t *ToTransformation) UpdateProcessingTime(id execute.DatasetID, pt execute.Time) error {
+	return t.d.UpdateProcessingTime(pt)
+}
+
+// Finish is called after the `to` flux function's transformation is done processing.
+func (t *ToTransformation) Finish(id execute.DatasetID, err error) {
+	if err != nil {
+		t.d.Finish(err)
+		return
+	}
+	err = t.buf.Flush(t.ctx)
+	t.d.Finish(err)
+}
+
+const (
```

Comment:
From here all the way down to `writeTable`, this code was copied from IDPE to here so it can be reused in both places.